Dockerfile

```diff
@@ -1,30 +0,0 @@
-# Builder image
```

Comment:
I understand the move to kos, which makes things way easier.

I just want to make aware of one drawback. Users can't build the dockercontainers by themselves anymore. Manual builds for testing use then a different container than the released ones. For testing in the pipeline we would use `docker.io/golang:1.21-bookworm` as basic and the kos settings would use `gcr.io/distroless/base-debian11`.

This just reminds me of the discussion that distroless containers are to "complicated" to maintain.

Reply:
This is a good point, for the time being, I am willing to give the distroless part another chance, and then keep the `Dockerfile.debug` around for "build your own".

I'm not sure if we can make the goreleaser push containers if there is no release, but that would be useful. 

Goreleaser supports a docker mode which is more traditional, but it requires more stuff, I would prefer to try `ko` first and then potentially switch to that if its not working out.

> This just reminds me of the discussion that distroless containers are to "complicated" to maintain.
I appreciate this comment ;), luckily I've right now forgotten all the pain points and will try this and then curse in the future. 

let me know what you think.

Reply:
We can try it of course. 

I would also add the debug image of distroless, as a lot of people use stuff like portainer and so and they normally want to have a shell inside the container. This is the same we did before we got rid of the old distroless containers. Furthermore the question is if ko can create our folders inside the container. As we had the problem beforehand: #1417 

So a bit of a general blueprint:

1. Container with distroless-debian12
2. Container with distroless-debian12:debug

I guess testing the whole thing on a fork with "fake releases" would be our best option to get this thing working and tested before merging into main. 

I hope I didn't reminded you of too much pain, because I'm still a huge fan of distroless images.

Reply:
hehehe, no worries, I've added debug but based on regular debian image as a compromise for now. Would you have time to help me test it?

Reply:
or could merge it and test it as part of the next alpha.